### ISSUE TITLE
Is photometric loss a l2 loss with sqrt or not?

### ISSUE THREAD

**Issue by havefunbb** [microsoft/Deep3DFaceReconstruction#21]
In your paper, the photometric loss is a l2 loss with sqrt operation. But when computing gradients, sqrt operation always cause NAN. So, I just want to make sure if the photometric loss is a l2 loss with sqrt? If it is, can you give some suggestion on how to deal with the NAN situation.

**Comment by YuDeng** [comment]
It is a l2 loss with sqrt (L2,1 loss). I did not encounter with NAN during my training process. You can try to add a small number, e.g. 1e-5 inside sqrt to avoid NAN.

**Comment by havefunbb** [comment]
Maybe it's because of the initialization of the last full connect layer, when I tried again, NAN just disappeared. Thank you all the same.

**Comment by hangon666** [comment]
@havefunbb 
Hello, could you give me some advice on how to reimplenment the photometric loss function?
